### PR TITLE
Cumulus 653 Test for Editing Rules

### PR DESCRIPTION
**Pull request by kkelly51**

### CODE REVIEW

File: app/scripts/components/rules/edit.js [hunk 1]
```diff
@@ -19,14 +19,15 @@ const EditRule = React.createClass({
   },
 
   render: function () {
-    const { ruleName } = this.props.params;
+    const { params, rules } = this.props;
```
Comment (markdboyd) [inline app/scripts/components/rules/edit.js]: It doesn't seem like you're using the value of `params` anywhere? If so, you can remove it

File: cypress/integration/rules_spec.js [hunk 2]
```diff
@@ -106,6 +106,35 @@ describe('Rules page', () => {
         });
     });
 
+    it('editing a rule and returning to the rules page should show the new changes', () => {
+      cy.visit('/#/rules');
+      cy.contains('table tbody tr a', testRuleName)
+      .and('have.attr', 'href', `#/rules/rule/${testRuleName}`).click();
```
Comment (markdboyd) [inline cypress/integration/rules_spec.js]: Can we update the indentation here for readability?

```suggestion
cy.contains('table tbody tr a', testRuleName)
  .and('have.attr', 'href', `#/rules/rule/${testRuleName}`)
  .click();
```

File: cypress/integration/rules_spec.js [hunk 3]
```diff
@@ -106,6 +106,35 @@ describe('Rules page', () => {
...
+      cy.contains('.heading--large', `Edit ${testRuleName}`);
+
+      // update collection and submit
+      const provider = 'newProvider';
+      cy.editJsonTextarea({ data: { provider }, update: true });
+      cy.contains('form input', 'Submit').click();
+
+      cy.contains('.heading--large', testRuleName);
+      cy.get('.metadata__details')
+      .within(() => {
```
Comment (markdboyd) [inline cypress/integration/rules_spec.js]: Can we move over the indentation by two spaces for readability?

File: app/scripts/components/rules/edit.js (hunk 1 above)
Comment (kkelly51) [reply app/scripts/components/rules/edit.js]: I'm using params in the next line (23) to pull out the ruleName.

File: cypress/integration/rules_spec.js [hunk 4]
```diff
@@ -106,6 +106,35 @@ describe('Rules page', () => {
...
+        cy.contains('Provider').next().should('have.text', 'newProvider');
+      });
+
+      cy.contains('a', 'Back to Rules').click();
+      cy.contains('.heading--large', 'Rule Overview');
+      cy.contains('table tr', testRuleName)
+      .within(() => {
```
[the remark this answers is not in the document]
Comment (kkelly51) [reply cypress/integration/rules_spec.js]: changed

File: cypress/integration/rules_spec.js (hunk 3 above)
Comment (kkelly51) [reply cypress/integration/rules_spec.js]: changed

File: cypress/integration/rules_spec.js [hunk 5]
```diff
@@ -106,6 +106,35 @@ describe('Rules page', () => {
...
+
+      // update collection and submit
+      const provider = 'newProvider';
+      cy.editJsonTextarea({ data: { provider }, update: true });
+      cy.contains('form input', 'Submit').click();
+
+      cy.contains('.heading--large', testRuleName);
+      cy.get('.metadata__details')
+      .within(() => {
+        cy.contains('Provider').next().should('have.text', 'newProvider');
```
[the remark this answers is not in the document]
Comment (kkelly51) [reply cypress/integration/rules_spec.js]: changed

File: cypress/integration/rules_spec.js (hunk 2 above)
Comment (kkelly51) [reply cypress/integration/rules_spec.js]: changed

File: app/scripts/components/rules/edit.js (hunk 1 above)
Comment (markdboyd) [reply app/scripts/components/rules/edit.js]: Not sure how I missed that. My bad

File: cypress/integration/rules_spec.js [hunk 6]
```diff
@@ -106,6 +106,36 @@ describe('Rules page', () => {
...
+      cy.visit('/#/rules');
+      cy.contains('table tbody tr a', testRuleName)
+        .and('have.attr', 'href', `#/rules/rule/${testRuleName}`)
+        .click();
+
+      cy.contains('.heading--large', testRuleName);
+      cy.contains('.button--small', 'Edit').click();
+      cy.contains('.heading--large', `Edit ${testRuleName}`);
+
+      // update collection and submit
```
Comment (jennyhliu) [inline cypress/integration/rules_spec.js]: update 'rule'

**Pull request merged**